lib/spack/spack/fetch_strategy.py

```diff
@@ -211,6 +216,9 @@ def fetch(self):
 
         if not self.archive_file:
             raise FailedDownloadError(self.url)
+        elif self.digest:
+            shutil.copy(self.archive_file, spack.cache_path)
```

Comment:
I've thought about implementing this before and I think it would be really useful if the cache itself were a valid spack mirror, because users could then easily copy it over to another network and put it in a shared location for general use.

You can use `spack.mirror.mirror_archive_path` for that -- it'll give you a unique location inside a spack mirror where a particular archive goes, so you can use that instead of the fetch basename.  If you just use the fetch URL basename, you run the risk of having two poorly named archives conflict w/each other.

Instead of using `shutil.copy` here, you can use `FetchStrategy.archive()`, which handles archiving source repositories.